pyopencl/algorithm.py

```diff
@@ -29,13 +29,19 @@
 OTHER DEALINGS IN THE SOFTWARE.
 """
 
+from dataclasses import dataclass
```

Comment:
The compat target for pyopencl is nominally still Py3.6, so this needs a conditional dependency in `setup.py`.

Reply:
Right! Added in [f9e31f4](https://github.com/inducer/pyopencl/pull/571/commits/f9e31f41439c9a7fdfc0ae8476a05f4d8388e125).